### PR TITLE
Publish docs on GitHub Pages.

### LINKED ISSUE #184: Release docs to Github pages

### PR DESCRIPTION
**Pull request by sabyadi**
Please see commit message.

Closes #184.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -204,7 +204,8 @@ test: _check_version develop_$(pymn).done
 
 .PHONY:	sanity
 sanity: _check_version develop_$(pymn).done
-	tar -rf $(sanity_tar_file) --exclude=tmp_workspace.tar --exclude=$(sanity_dir1) .
+	tar -rf $(sanity_tar_file) --exclude=tmp_workspace.tar --exclude=$(doc_build_dir) --exclude=$(sanity_dir1) --exclude=$(doc_source_dir)  .
```
Comment (andy-maier) [inline Makefile:207]: In another change, I have removed the exclusion of doc_source_dir, in order to get the manually maintained .rst files checked. If the generated .rst files create issues, we can exclude doc_source_dir/modules, but for now they check ok.

**Comment by andy-maier** [comment]
So far I have ome comment, but I want to try it out as well, particularly the moving of the subtree.

File: Makefile (hunk 1 above)
Comment (sabyadi) [reply Makefile]: Done.

File: Makefile [hunk 2]
```diff
@@ -280,4 +281,10 @@ ifneq ($(doc_build),true)
 	@echo "makefile: Warning: Skipping docs build on Python $(python_m_n_version)"
 else
 	sphinx-build -M html $(sphinx_opts) $(doc_source_dir) $(doc_build_dir)
+	# move the files one level up for the GitHub pages
+	rm -rf $(doc_build_dir)/doctrees
+	cp -R $(doc_build_dir)/html/* $(doc_build_dir)/
+	rm -rf $(doc_build_dir)/html
+	# disable GitHub pages jekyll pre-processing
+	touch $(doc_build_dir)/.nojekyll
```
Comment (andy-maier) [inline Makefile:289]: I'm struggling with the move. Two concrete issues are:
* The target `$(doc_build_dir)/html/index.html` is moved one level up so `make docs` is never in a complete state and rebuilds it every time. So if we stick with the move, the target file needs to be changed to `$(doc_build_dir)/index.html`.
* The PR puts the .html files into the repo in their location before the move. If we keep them in the repo, they should be in the repo with their locations after the move.

**Review by andy-maier — CHANGES_REQUESTED** [review]
see comment

File: Makefile [hunk 3]
```diff
@@ -111,8 +111,8 @@ sanity_dir1 := tmp_sanity
 sanity_tar_file := tmp_workspace.tar
 
 # Directories for documentation
-doc_source_dir := docs/source
-doc_build_dir := docs/build
+doc_source_dir := docs_source
+doc_build_dir := docs
```
Comment (andy-maier) [inline Makefile:115]: I just found that `make clobber` removes `$(doc_build_dir)`, but if we want to keep the built docs files in the repo, it should not be removed in `make clobber`

File: Makefile (hunk 2 above)
Comment (sabyadi) [reply Makefile:289]: The GitHub Pages want to have the documentation in the /docs folder, so this is why I moved the sources into the /docs_source and modified the generation to generate into the /docs. The copy one love up is due to the fact that sphinx generated the html files in the html subfolder. I've corrected the target.

File: Makefile (hunk 3 above)
Comment (sabyadi) [reply Makefile:115]: Fixed.

Comment (vkathir82) [reply Makefile:115]: Yes, I have added that in my PR earlier to clean generated html pages and it needs to be kept there to publish the docs

**Comment by andy-maier** [comment]
Rebased and resolved merge conflicts. Fixed docs/html in two more cases. Rebuilt docs.

**Review by andy-maier — APPROVED** [review]

**Pull request merged**